crew/op_connect.go

```diff
@@ -516,18 +529,48 @@ func (op *ConnectOp) HandleStop(err *terminal.Error) (errorToSend *terminal.Erro
 	// Cancel workers.
 	op.cancelCtx()
 
-	// Avoid connecting to destination via this Hub if the was a connection
-	// error and no data was received.
-	if op.entry && // On clients only.
-		err.Is(terminal.ErrConnectionError) &&
-		op.outgoingTraffic.Load() == 0 {
-		// Only if no data was received (ie. sent to local application).
-		op.tunnel.avoidDestinationHub()
-	}
+	// Special client-side handling.
+	if op.entry {
+		// Mark the connection as failed if there was an error and no data was sent to the app yet.
+		if err.IsError() && op.outgoingTraffic.Load() == 0 {
+			// Set connection to failed and save it to propagate the update.
```

Comment:
`err.IsError()` is not very intuitive. I guess either it should be an error or not (i.e. `err == nil`)?